Model optimizataion documentation update (#11072)
openvinotoolkit/openvino: tools/pot/docs/AccuracyAwareQuantizationUsage.md

# Quantizing Model with Accuracy Control{#pot_accuracyaware_usage}

@sphinxdirective

.. toctree::
   :maxdepth: 1
   :hidden:

   AccuracyAwareQuantization Method <accuracy_aware_README>

@endsphinxdirective

## Introduction

> This document assumes that you already tried [Default Quantization](@ref pot_default_quantization_usage) for the same model. In case when it introduces a significant accuracy degradation, Accuracy-aware Quantization algorithm can be used to remain accuracy within the pre-defined range. This may cause a 

Reviewer: ```suggestion
This document assumes that you already tried [Default Quantization](@ref pot_default_quantization_usage) for the same model. In case when it introduces a significant accuracy degradation, the Accuracy-aware Quantization algorithm can be used to remain accuracy within the pre-defined range. This may cause a 
```

degradation of performance in comparison to [Default Quantization](@ref pot_default_quantization_usage) algorithm because some layers can be reverted back to the original precision.

> **NOTE**: In case of GNA `target_device`, the Accuracy-aware Quantization algorithm behavior is different. It is searching for the best configuration selecting between INT8 and INT16 precisions for weights of each layer. The algorithm works for the `performance` preset only. For the `accuracy` preset, this algorithm is not helpful since the whole model is already in INT16 precision.

> A script for Accuracy-aware Quantization should include four steps:

Reviewer: ```suggestion
A script for Accuracy-aware Quantization includes four steps:
```

1. Prepare data and dataset interface
2. Define accuracy metric
3. Select quantization parameters
4. Define and run quantization process

## Prepare data and dataset interface
This step is the same as in the case of [Default Quantization](@ref pot_default_quantization_usage). The only difference is that `__getitem__()` method should return `(data, annotation)` or `(data, annotation, metadata)` where `annotation` is required and it format should correspond to the expectations of the `Metric` class. `metadata` is an optional field that can be used to store additional information required for post-processing.

## Define accuracy metric

> In order to control accuracy during the optimization a `openvino.tools.pot.Metric` interface should be implemented. Each implementaion should override the following properties:

Reviewer: ```suggestion
To control accuracy during the optimization a `openvino.tools.pot.Metric` interface should be implemented. Each implementation should override the following properties:
```

- `value` - returns the accuracy metric value for the last model output in a format of `Dict[str, numpy.array]`.
- `avg_value` - returns the average accuracy metric over collected model results in a format of `Dict[str, numpy.array]`.

> - `higher_better` should return `True` if a higher value of the metric corresponds to a better performance, otherwise, returns `False`. Default implementation returns `True`.

Reviewer: ```suggestion
- `higher_better` should return `True` if a higher value of the metric corresponds to better performance, otherwise, returns `False`. Default implementation returns `True`.
```

and methods:

> - `update(output, annotation)` - calculates and updates the accuracy metric value using last model output and annotation. The model ouput and annotation should be passed in this method. It should also contain the model specific post-processing in case if the model returns the raw output.

Reviewer: ```suggestion
- `update(output, annotation)` - calculates and updates the accuracy metric value using the last model output and annotation. The model output and annotation should be passed in this method. It should also contain the model-specific post-processing in case the model returns the raw output.
```

            'maximal_drop': 0.02
        },
    }
]

# Step 1: implement and create user's data loader
data_loader = UserDataLoader()

# Step 2: implement and create user's data loader
metric = Accuracy()

# Step 3: load model
model = load_model(model_config=model_config)

# Step 4: Initialize the engine for metric calculation and statistics collection.
engine = IEEngine(config=engine_config, data_loader=data_loader, metric=metric)

# Step 5: Create a pipeline of compression algorithms and run it.
pipeline = create_pipeline(algorithms, engine)
compressed_model = pipeline.run(model=model)

# Step 6 (Optional): Compress model weights to quantized precision
#                    in order to reduce the size of the final .bin file.
compress_model_weights(compressed_model)

# Step 7: Save the compressed model to the desired path.
# Set save_path to the directory where the model should be saved
compressed_model_paths = save_model(
    model=compressed_model,
    save_path="optimized_model",
    model_name="optimized_model",
)

# Step 8 (Optional): Evaluate the compressed model. Print the results.
metric_results = pipeline.evaluate(compressed_model)
```

> It is worth noting that now `evaluate` method that can compute accuracy on demand is also available in the `Pipeline` object.

Reviewer: ```suggestion
It is worth noting that now the `evaluate` method that can compute accuracy on demand is also available in the `Pipeline` object.
```

> In case when Accuracy-aware Quantization does not allow to achieve the desired accuracy-performance trade-off, it is recommended to try Quantization-aware Training from [NNCF](@ref docs_nncf_introduction).

Reviewer: ```suggestion
In case when Accuracy-aware Quantization does not allow achieving the desired accuracy-performance trade-off, it is recommended to try Quantization-aware Training from [NNCF](@ref docs_nncf_introduction).
```
